### PR TITLE
Make sure to sanitize pinned plugins in plugin_check (Fixes #192)

### PR DESCRIPTION
**Pull request by jedahan**
I had to do this to fix `+rustup` in my config

### CODE REVIEW

File: lib/plugin.zsh [hunk 1]
```diff
@@ -72,7 +72,7 @@ geometry_plugin_register() {
     return 1
   fi
 
-  if geometry_prompt_${plugin#+}_setup $ctx; then
+  if $plugin_setup_function $ctx; then
```
Comment (desyncr) [inline lib/plugin.zsh:75]: Isn't this sanitized on [this](https://github.com/geometry-zsh/geometry/pull/194/files#diff-fa15ebb02da94453c62e6471145db51aR69) line?

Comment (jedahan) [reply lib/plugin.zsh:75]: this line isn't the fix, this was just to reduce redundancy since we have access to the local anyway

Comment (desyncr) [reply lib/plugin.zsh:75]: oh I see now! :+1:

**Review by desyncr — APPROVED** [review]

**Review by frm — APPROVED** [review]

**Comment by desyncr** [comment]
I tried it and I see no breaking changes. What's a test case?

**Comment by desyncr** [comment]
btw, my configs:

```
GEOMETRY_PROMPT_PLUGINS_SECONDARY=(exec_time jobs +node git +hostname)
GEOMETRY_PROMPT_PLUGINS_PRIMARY=(path)
```

**Comment by jedahan** [comment]
On a fresh install of geometry, minimal zshrc with `GEOMETRY_PROMPT_PLUGINS_SECONDARY=(+rustup)` before I would get errors 'rustup plugin not found'. Now I don't.

**Comment by desyncr** [comment]
Mergin then!

**Pull request merged**